data/src/test/java/com/example/data/DataApplicationTests.java

```diff
@@ -0,0 +1,13 @@
+package com.example.data;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class DataApplicationTests {
```

Comment:
I would recommend to move away into separate sourceSet all heavy integration tests. 
Try to create a new sourceSet and task `integration-test` and also try to add the integration test to test out your controller. That would be useful.

Here should be unit tests only. 
The unit test should run really fast and test only tested unit. Loading the Spring Context and loading many other units breaks that concept.

Your unit test must import just the tested class and all the dependencies must be just mocked.
Check it out what is `test pyramid`.